qa/rpc-tests/segwit.py

```diff
@@ -211,7 +211,18 @@ def run_test(self):
         block = self.nodes[2].generate(1) #block 432 (first block with new rules; 432 = 144 * 3)
         sync_blocks(self.nodes)
         assert_equal(len(self.nodes[2].getrawmempool()), 0)
-        assert_equal(len(self.nodes[2].getblock(block[0])["tx"]), 5)
+        segwit_tx_list = self.nodes[2].getblock(block[0])["tx"]
+        assert_equal(len(segwit_tx_list), 5)
+
+        print("Verify block and transaction serialization rpcs return differing serializations depending on rpc serialization flag")
+        # Note: node1 has version 2, which is simply >0 and will catch future upgrades in tests
+        assert(self.nodes[2].getblock(block[0], False) !=  self.nodes[0].getblock(block[0], False))
+        assert(self.nodes[1].getblock(block[0], False) ==  self.nodes[2].getblock(block[0], False))
+        for i in range(len(segwit_tx_list)):
+            assert(self.nodes[2].getrawtransaction(segwit_tx_list[i]) != self.nodes[0].getrawtransaction(segwit_tx_list[i]))
+            assert(self.nodes[1].getrawtransaction(segwit_tx_list[i], 0) == self.nodes[2].getrawtransaction(segwit_tx_list[i]))
+            assert(self.nodes[0].getrawtransaction(segwit_tx_list[i]) != self.nodes[2].gettransaction(segwit_tx_list[i])["hex"])
+            assert(self.nodes[1].getrawtransaction(segwit_tx_list[i]) == self.nodes[2].gettransaction(segwit_tx_list[i])["hex"])
```

Comment:
Perhaps the test should explicitly test the correctness of the serializations, rather than just comparing output against each other?  (Maybe just the node0/legacy-serialization case, since I presume something would break if the `rpcserializationversion=1` handling were busted.)